### ISSUE TITLE
Update ADS datasets following BEP release

### ISSUE THREAD

**Issue by mariehbourget** [neuropoly/axondeepseg#603]
The Microscopy BEP was integrated in BIDS and can be previewed [here](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/10-microscopy.html).

It will be part of the BIDS 1.7.0 release coming soon (**EDIT: released on Feb 15th**), without changes ~except for a JSON files for photos files that we don't use (AFAIK). This part is not confirmed yet but can be previewed for reference [here](https://bids-specification--1000.org.readthedocs.build/en/1000/04-modality-specific-files/10-microscopy.html#photos-of-the-samples-_photoextension).~ **EDIT: the JSON photo file was not be part of the 1.7.0 release, so it can be ignored for now.**

Here are the updates that need to be done to our ADS datasets, in order to be fully compliant and before we can complete the BEP implementation in ivadomed (ref: https://github.com/ivadomed/ivadomed/pull/1025).
Note: this will not affect the behavior of segmentation with the new ADS v4, but is necessary for training new models in ivadomed.

#### SEM dataset

- [x]  Up to date with version 1.7.0 and available [here](https://github.com/axondeepseg/data_axondeepseg_sem)

#### BF dataset

- [x] Up to date with version 1.7.0 on duke
- [x] Need to be uploaded to git-annex (ref: https://github.com/neuropoly/data-management/issues/152)

#### TEM dataset
- [x] Corrected "fuzzy mask" to update on git-annex (ref: https://github.com/neuropoly/data-management/issues/135#issuecomment-1020293016)
- [x] Update dataset to BIDS 1.7.0:
   - [x] In dataset_description.json, update `"BIDSVersion"` to `"1.7.0"`.
   - [x] In derivatives/labels/dataset_description.json, update `"BIDSVersion"` to `"1.7.0"` and add the field `"GeneratedBy": [{"Name": "Axon and myelin manual segmentation labels"}]` (this is also related to pybids in https://github.com/ivadomed/ivadomed/pull/994).
   - [x] All `microscopy` folders must be renamed `micr`.
   - [x] In JSON sidecar in each "sub-XX/micr" folder: remove the `"FieldOfView"` field and value (not part of the spec anymore), add the field `"PixelSizeUnits": "um"` (now required) and replace `"Environment": "exvivo"` by `"SampleEnvironment": "ex vivo"` (key and value changed).

#### Wakehealth dataset (git-annex)
- [x] Update dataset to BIDS 1.7.0, I'm less familiar with this dataset but it was curated at the same time as the TEM so the changes should be about the same, refer to the spec if other metadata field are present.

@hermancollin, let me know of you are still available to help with this issue. Thanks!

**Comment by hermancollin** [comment]
@mariehbourget yes I can help I'll let you know if anything comes up. Also, is there a reason for the SEM dataset being the only one uploaded on github? I thought git-annex was the new convention.

**Comment by mariehbourget** [comment]
> @mariehbourget yes I can help I'll let you know if anything comes up. Also, is there a reason for the SEM dataset being the only one uploaded on github? I thought git-annex was the new convention.

Thank you @hermancollin, your help is much appreciated!

For the SEM dataset, we have it on github since it is fairly small and is also used as a public example datasets for Microscopy-BIDS.

> Also, most images in the wakehealth dataset don't have masks. Should I discard the unlabelled images from the dataset or keep them in?

We should keep everything, there will simply be less derivatives sub-folder than sources sub-folder.
If needed, you can have a "full" dataset and a "training" dataset. This is what I did for the BF datasets. In that specific case, it was because we had different versions/crops of the same images and splitting in 2 datasets was clearer.
I would recommend also adding a README file in the dataset with a description.

**Comment by mariehbourget** [comment]
Update: As of today, it seems that the additional JSON for photo files will not be part of the BIDS 1.7.0 release, I updated the first comment.

**Comment by hermancollin** [comment]
@mariehbourget If you want to take a look, I updated the wakehealth dataset on git-annex on the branch `ac/bids-update`. I will add more content to the README but the bids part should be up-to-date. Also, I'll upload a separate training dataset, as you suggested.

On git-annex, the dataset is simply named `wakehealth`. Should we change it for something more like *data_axondeepseg_something*, like the others? I will probably rename the dataset anyway to differentiate _source and _training.

**Comment by mariehbourget** [comment]
> @mariehbourget If you want to take a look, I updated the wakehealth dataset on git-annex on the branch `ac/bids-update`. I will add more content to the README but the bids part should be up-to-date. Also, I'll upload a separate training dataset, as you suggested.

Thanks! I'll take a look a the datasets and try the bids-validator on them (unfortunately not available on the web yet but I have a local installation).

> On git-annex, the dataset is simply named `wakehealth`. Should we change it for something more like _data_axondeepseg_something_, like the others? I will probably rename the dataset anyway to differentiate _source and _training.

Yes, that's a great idea and will keep everything together! I think `data_axondeepseg_wakeheath_training` and `source` would do.

**Comment by mariehbourget** [comment]
💯 The wakehealth dataset pass the validator without errors (only some "normal" warnings).

If you are interested, you can install it locally from the master branch and run it in command-line with these instructions: https://docs.google.com/document/d/1wksxQq0lGduBEb90VZBA_FHnDoiddIvGeQEMLvu0ng8/edit#heading=h.714kb1vqkpkc
Otherwise, I can check them for you, it's not a problem.

Thinking about it, I think we should add the original NDPI files in the dataset, in a `sourcedata` folder. See `data_axondeepseg_bf_source/sourcedata` for reference.

**Comment by hermancollin** [comment]
@mariehbourget The `data_axondeepseg_wakehealth_source` dataset is now updated with its original NDPI files in sourcedata.

I am trying to pass the bids-validator test for the training dataset but I have a problem. We have 4 ROIs extracted from images with names such as *sub-21133_sample-21133_chunk-4_BF.tif*. Previously, I used the `desc` entity to name them like so:
- *sub-21133_sample-21133_chunk-4_desc-crop1_BF.tif*
- *sub-21133_sample-21133_chunk-4_desc-crop2_BF.tif*
- ...

Now, using the appropriate `acq` entity, it looks something like this:
- *sub-21133_sample-21133_chunk-4_acq-roi1_BF.tif*
- *sub-21133_sample-21133_chunk-4_acq-roi2_BF.tif*
- ...

But I get an error:
```
[ERR] Files with such naming scheme are not part of BIDS specification.
```

I feel like the problem is that the `acq` entity comes after `chunk`. Also, in `data_axondeepseg_bf_training`, you used chunks to identity the ROIs but I can't use it for this purpose because we would lose the information about the chunk it came from (chunk-4 in this example). Maybe I could simply specify what image it came from and use the chunk entity to enumerate the ROIs?

**Comment by mariehbourget** [comment]
> Previously, I used the `desc` entity to name

Oh yeah, I remember discussing this last summer.
Was `desc` working for the training in ivadomed?

If yes, it is because we force index microscopy files (skipping the validator).
This is a bit unclear because ideally we would used `desc` but `desc` is only acceptable for derivatives (as per BIDS). The issue here is that we use the ROI data as "raw data" and not "derivatives" in the ivadomed pipeline (so we are kind of in a "corner" case).

We have an issue about the naming of derivatives dataset in ivadomed, but did not get around a solution yet: https://github.com/ivadomed/ivadomed/issues/860

My recommendation for now would be to:
1) Keep the same name as the one used for your training (so we can reproduce it), IIUC with `desc`.
2) Document this in the README of the dataset.
3) Open a separate issue about the naming and failing of the validator and linked it to the ivadomed issue so we can look further into this at a later time i.e. if we integrate microscopy to pybids and not force index the files, it will fail.

> I feel like the problem is that the `acq` entity comes after `chunk`.

Yes, the order is important, if you put `acq` before `chunk`, it would work. See order [here](https://bids-specification.readthedocs.io/en/latest/04-modality-specific-files/10-microscopy.html#microscopy-imaging-data).

> Also, in `data_axondeepseg_bf_training`, you used chunks to identity the ROIs but I can't use it for this purpose because we would lose the information about the chunk it came from (chunk-4 in this example). Maybe I could simply specify what image it came from and use the chunk entity to enumerate the ROIs?

In the case we decide to not use `desc`, I feel it would be better to keep the "link" to the original chunk and use the `acq` label for ROI. But we may want to revisit that as well to make `bf` and `wakehealth` more consistent.

**Comment by hermancollin** [comment]
Ok. I have uploaded the training dataset on git-annex (*datasets/data_axondeepseg_wakehealth_training*). For now, it is consistent with the `bf` dataset (I used `acq` to label ROIs and specified the source chunk in the sourcedata folder). I will roll it back to the version I used for training in ivadomed (with `desc` entities) but I will keep the current version in a branch in case we decide to make both bright-field datasets more consistent.

Also, since I finally figured how to create a git-annex repo, I could upload both `data_axondeepseg_bf_source` and its training counterpart if Nick doesn't answer.

**Comment by mariehbourget** [comment]
Update on the validator:
There was a release ([1.9.0](https://github.com/bids-standard/bids-validator/releases/tag/v1.9.0)) of the validator 6 days ago.
This means that we can now validate our microscopy datasets directly on the web version with Chrome or Filefox here: http://bids-standard.github.io/bids-validator/ 🎉 

@hermancollin, are you still available to update the TEM dataset?

**Comment by hermancollin** [comment]
> Update on the validator: There was a release ([1.9.0](https://github.com/bids-standard/bids-validator/releases/tag/v1.9.0)) of the validator 6 days ago. This means that we can now validate our microscopy datasets directly on the web version with Chrome or Filefox here: http://bids-standard.github.io/bids-validator/ tada

Great news! This will be a lot simpler to use

> @hermancollin, are you still available to update the TEM dataset?

Ah yes forgot about that one. Sure, I can take care of the TEM one and also the fuzzy mask thing but I would need writing access on the dataset. I'll ask around

**Comment by hermancollin** [comment]
@mariehbourget btw was the default TEM model trained with the weird *sub-nyuMouse15_sample-0005* masks? The model README points to commit c778a3 which still contains the wrong masks.

The TEM dataset is now up to date on git-annex and the problematic masks should be fixed as well.

**Comment by mariehbourget** [comment]
> @mariehbourget btw was the default TEM model trained with the weird _sub-nyuMouse15_sample-0005_ masks? The model README points to commit c778a3 which still contains the wrong masks.

Oh yes, the model was trained with the weird mask 😬. It should not have too much impact on the results because the dataset is pretty large so there were plenty of "good" examples.

> The TEM dataset is now up to date on git-annex and the problematic masks should be fixed as well.

Thank you again for your help! I'll close this issue and we'll follow-up on the wakehealth dataset in #605.
